### DISCUSSION TITLE
DBC Shifted Bits

### DISCUSSION THREAD

**Discussion by jamiejones85** [collin80/SavvyCAN#309] [Q&A] [answered]
Hi 

I'm looking at creating a DBC file for the following rule.

`  voltage = uint16_t(rxBuf[1] >> 4) + uint16_t(rxBuf[2] << 4) + 1000;`

Is this possbile?

**Comment by collin80** [comment, marked as answer]
Yes, I believe what you're looking for is a starting bit of 12, a bit length of 12, little endian mode. This causes the upper 4 bits of byte 1 to be the low 4 bits in the signal and then the 8 bits of byte 2 to be the upper 8 bits of the signal. You also need a bias value of 1000. 

It should look something like this in SavvyCAN:
\
![Screenshot_20210309_111036](https://user-images.githubusercontent.com/2145125/110501459-33743380-80c8-11eb-9e5e-74ea9d0ed74c.png)

**Reply by jamiejones85** [reply]
Awesome, that seems to work. 
I'm trying to create a DBC file for VW GTE battery modules if anyone has any use for it https://github.com/jamiejones85/DBC-files/blob/master/VW-GTE-HV-Battery.dbc
